client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java

```diff
@@ -553,16 +576,36 @@ public void incPartitionFetchFailure(int stageAttempt, int partition) {
           });
     }
 
-    public int getPartitionFetchFailureNum(int stageAttempt, int partition) {
+    public boolean getPartitionFetchFailureNum(
+        int stageAttempt, int partition, RssShuffleManagerInterface shuffleManager) {
       return withReadLock(
           () -> {
             if (this.stageAttempt != stageAttempt) {
-              return 0;
+              return false;
             } else {
-              return this.partitions[partition];
+              if (this.partitions[partition] >= shuffleManager.getMaxFetchFailures()) {
+                return true;
+              } else {
+                return false;
+              }
             }
           });
     }
+
+    public void clearedMapTrackerBlock(boolean isCleared) {
+      withWriteLock(
+          () -> {
+            this.isClearedMapTrackerBlock = isCleared;
+            return null;
+          });
+    }
+
+    public boolean isClearedMapTrackerBlock() {
```

Comment:
`isClearedMapTrackerBlock` -> `hasClearedMapTrackerBlock`

Reply:
The above has been changed.